common/src/View/ViewEditor.cpp

```diff
@@ -220,7 +221,7 @@ namespace TrenchBroom {
         }
 
         void ViewEditor::bindObservers() {
-            auto document = lock(m_document);
+            auto document = kdl::mem_lock(m_document);
```

Comment:
While I do think it would be an improvement to drop kdl::mem_lock, I understand it's a matter of taste (and there's enough in this PR so I wouldn't suggest changing it now anyway) :)

Generally I have the same problem with this as `kdl::do_before` which just executes a lambda immediately - library code which essentially does "nothing" still has a cost in terms of codebase complexity, and familiarity - I get to know my way around std:: so I remember what things do without looking up docs, but this is less likely for `kdl`. This applies even more for people who are new to the codebase.

Reply:
`invoke_before` is in and of itself not very useful and is only there for reasons of symmetry. I'm ok with removing that for the reasons given. `mem_lock` etc. do have a benefit: you can treat shared / weak pointers in the same way.